src/pip/_internal/operations/prepare.py

```diff
@@ -376,109 +376,104 @@ def _download_should_save(self):
             "Could not find or access download directory '{}'"
             .format(self.download_dir))
 
-    def prepare_linked_requirement(
-        self,
-        req,  # type: InstallRequirement
-        parallel_builds=False,  # type: bool
-    ):
-        # type: (...) -> AbstractDistribution
-        """Prepare a requirement that would be obtained from req.link
-        """
-        assert req.link
-        link = req.link
-
-        # TODO: Breakup into smaller functions
-        if link.scheme == 'file':
-            path = link.file_path
+    def _log_preparing_link(self, req):
+        # type: (InstallRequirement) -> None
+        """Log the way the link prepared."""
+        if req.link.is_file:
+            path = req.link.file_path
             logger.info('Processing %s', display_path(path))
         else:
             logger.info('Collecting %s', req.req or req)
 
-        download_dir = self.download_dir
-        if link.is_wheel and self.wheel_download_dir:
-            # when doing 'pip wheel` we download wheels to a
-            # dedicated dir.
-            download_dir = self.wheel_download_dir
-
-        if link.is_wheel:
+    def _ensure_link_req_src_dir(self, req, download_dir, parallel_builds):
+        # type: (InstallRequirement, Optional[str], bool) -> None
+        """Ensure source_dir of a linked InstallRequirement."""
+        # Since source_dir is only set for editable requirements.
+        if req.link.is_wheel:
             if download_dir:
                 # When downloading, we only unpack wheels to get
                 # metadata.
                 autodelete_unpacked = True
             else:
-                # When installing a wheel, we use the unpacked
-                # wheel.
+                # When installing a wheel, we use the unpacked wheel.
                 autodelete_unpacked = False
```

Comment:
```python
# When downloading, we only unpack wheels to get metadata, 
# but when installing we use the unpacked wheel
autodelete_unpacked = True if download_dir else False
```

Reply:
I also prefer the shorter version, but let's wait for @pradyunsg to settle this: https://github.com/pypa/pip/pull/8411#discussion_r438144983

Reply:
Let's keep this as is. We can change this later if someone really thinks we should.